### PR TITLE
ci: update production deploy to only release affected apps

### PR DESCRIPTION
**Pull request by overbit**
Close: #6786 

## PR Details

Working example:
⚠️ the video was recorded before the renaming of the workflow to `Release Production`

https://github.com/amplication/amplication/assets/2861984/d75b4d2a-5336-4ffc-b525-e21698ab40d0

### Summary
🚀📝🛠️

This pull request improves the way the application version is generated and used in the client and server packages. It removes the `version` field from the root `package.json` file and uses GitHub and git tags to get the latest release and local versions. It also updates the scripts, the `.gitignore` files, and the Nx project configuration to reflect these changes.

> _We are the masters of `version.ts`_
> _We write the script that rules the rest_
> _We use the tags from GitHub and git_
> _We don't need the cache or the `nx-force-all` bit_

### Walkthrough
*  Remove `version` field from root `package.json` and replace it with dynamic scripts that fetch the latest release tag and local tag from GitHub and git, respectively, and write them to `version.ts` files in each package ([link](https://github.com/amplication/amplication/pull/6787/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L3), [link](https://github.com/amplication/amplication/pull/6787/files?diff=unified&w=0#diff-ccfb34be901a566500579fd99e6425f84aa48b2e09db513c41c200af1335cdb2L1-R3), [link](https://github.com/amplication/amplication/pull/6787/files?diff=unified&w=0#diff-9634303137442d185bc523324a75aef222e748c772138aacc82e00e1f8d3ba10L1-R6), [link](https://github.com/amplication/amplication/pull/6787/files?diff=unified&w=0#diff-9634303137442d185bc523324a75aef222e748c772138aacc82e00e1f8d3ba10L15-R28), [link](https://github.com/amplication/amplication/pull/6787/files?diff=unified&w=0#diff-7346c03dec9e219312da322832f888587b6efafce22e8e56f847660426e4562aR13-R15), [link](https://github.com/amplication/amplication/pull/6787/files?diff=unified&w=0#diff-c5b682550310d16ccc0484158ea019afa825a387eae2d7f90208883b7e54d055L37-R54), [link](https://github.com/amplication/amplication/pull/6787/files?diff=unified&w=0#diff-1a6841ef2d2a763c141422cb9431305c36a3ea36b399f97a37e144c0ff0e84bfR1-R36), [link](https://github.com/amplication/amplication/pull/6787/files?diff=unified&w=0#diff-8fa154a4b5b3c95beb7d5ea62044a3422121b74ed8ae66b36823aca1e5635b9cL5-R9), [link](https://github.com/amplication/amplication/pull/6787/files?diff=unified&w=0#diff-8fa154a4b5b3c95beb7d5ea62044a3422121b74ed8ae66b36823aca1e5635b9cR37))
* Downgrade `prisma-schema-dsl` dependency to fix compatibility issue with Prisma CLI ([link](https://github.com/amplication/amplication/pull/6787/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L146-R145))
* Remove `nx-force-all` option from `build` step of `deployment.production.yml` workflow, since cache is disabled by default in Nx ([link](https://github.com/amplication/amplication/pull/6787/files?diff=unified&w=0#diff-f30aca93fe86bb7f94310fa565b1c0a04c02cffe55d6e83c9998c59b987e2f55L164))

## PR Checklist

- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: .github/workflows/release.production.yml [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        description: Check this ONLY if you want to re-run the workflow for the same version.
+      nx-force-all:
+        type: boolean
+        description: Forces Nx to consider all projects as affected when running the update-version target.
+        default: true
+      draft-release:
+        type: boolean
+        description: Create a draft release instead of a published release.
+        required: false
+        default: false
```
Comment (overbit) [inline .github/workflows/release.production.yml:24]: Let's default to `true`

Comment (overbit) [inline .github/workflows/release.production.yml:24]:
```suggestion
        default: true
```

File: packages/amplication-server/src/util/version.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export const version = "1.8.9";
```
Comment (abrl91) [inline packages/amplication-server/src/util/version.ts:1]: Do we need to change it?

Comment (overbit) [reply packages/amplication-server/src/util/version.ts]: Done

File: .github/workflows/release.production.yml [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('package-lock.json') }}
+
+      - name: Install dependencies
+        if: ${{ steps.cache-node-modules.outputs.cache-hit != 'true' }}
+        run: npm ci
+        
+      - id: update-version-ts
+        name: Update apps version.ts
```
Comment (overbit) [inline .github/workflows/release.production.yml:95]:
```suggestion
      - name: Update apps version.ts
        id: update-version-ts
```

**Review by abrl91 — APPROVED** [review]

**Pull request merged**